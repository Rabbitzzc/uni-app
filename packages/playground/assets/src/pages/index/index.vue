<template>
    <view>
        <navigator url="/pages/sub/test/test">test</navigator>
        <test-com></test-com>
        <view>
            <view>3:{{invalidStatic}}</view>
            <image src="./static/invalid-static.jpg" />
        </view>
        <view>
            <view>4:{{logo}}</view>
            <image src="../../static/logo.png" />
        </view>
    </view>
</template>
<script>
    import invalidStatic from './static/invalid-static.jpg'
    import logo from '../../static/logo.png'
    export default {
        data() {
            return {
                invalidStatic,
                logo
            }
        }
    }
</script>