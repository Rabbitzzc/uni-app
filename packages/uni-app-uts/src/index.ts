import { AutoImportOptions } from '@dcloudio/uni-cli-shared'
import { initAndroid, initIOS } from './plugins'

export default (options: { autoImportOptions?: AutoImportOptions }) => {
  return process.env.UNI_UTS_PLATFORM === 'app-android'
    ? initAndroid(options)
    : initIOS()
}

export { genClassName } from './plugins/android/utils'

export { transformMain as transformVue } from './plugins/android/uvue/sfc/main'
