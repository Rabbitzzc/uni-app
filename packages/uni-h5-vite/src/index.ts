import path from 'path'
import {
  isVueSfcFile,
  uniCssScopedPlugin,
  UNI_EASYCOM_EXCLUDE,
  resolveUTSCompiler,
  uniUTSUVueJavaScriptPlugin,
  isAppVue,
} from '@dcloudio/uni-cli-shared'
import { uniH5Plugin } from './plugin'
import { uniCssPlugin } from './plugins/css'
import { uniEasycomPlugin } from './plugins/easycom'
import { uniInjectPlugin } from './plugins/inject'
import { uniMainJsPlugin } from './plugins/mainJs'
import { uniManifestJsonPlugin } from './plugins/manifestJson'
import { uniPagesJsonPlugin } from './plugins/pagesJson'
import { uniPostVuePlugin } from './plugins/postVue'
import { uniRenderjsPlugin } from './plugins/renderjs'
import { uniResolveIdPlugin } from './plugins/resolveId'
import { uniSetupPlugin } from './plugins/setup'
import { uniSSRPlugin } from './plugins/ssr'
import { uniPostSourceMapPlugin } from './plugins/sourcemap'
import * as vueCompilerDom from '@vue/compiler-dom'
import * as uniCliShared from '@dcloudio/uni-cli-shared'

export default [
  ...(process.env.UNI_APP_X === 'true'
    ? [
        uniUTSUVueJavaScriptPlugin(),
        resolveUTSCompiler().uts2js({
          inputDir: process.env.UNI_INPUT_DIR,
          version: process.env.UNI_COMPILER_VERSION,
          cacheRoot: path.resolve(
            process.env.UNI_APP_X_CACHE_DIR ||
              path.resolve(process.env.UNI_OUTPUT_DIR, '../.web'),
            '.uts/cache'
          ),
          modules: {
            vueCompilerDom,
            uniCliShared,
          },
        }),
      ]
    : []),
  uniEasycomPlugin({ exclude: UNI_EASYCOM_EXCLUDE }),
  uniCssScopedPlugin({
    filter: (id) => isVueSfcFile(id) && !isAppVue(id),
  }),
  uniResolveIdPlugin(),
  uniMainJsPlugin(),
  uniManifestJsonPlugin(),
  uniPagesJsonPlugin(),
  uniInjectPlugin(),
  uniCssPlugin(),
  uniSSRPlugin(),
  uniSetupPlugin(),
  uniRenderjsPlugin(),
  uniH5Plugin(),
  uniPostVuePlugin(),
  uniPostSourceMapPlugin(),
]
