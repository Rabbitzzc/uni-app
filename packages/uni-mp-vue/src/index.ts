import plugin from './plugin'
// @ts-expect-error
import { createVueApp } from 'vue'
export function createApp(rootComponent: unknown, rootProps = null) {
  rootComponent && ((rootComponent as any).mpType = 'app')
  return createVueApp(rootComponent, rootProps).use(plugin)
}
export const createSSRApp = createApp
export * from './helpers'
// @ts-expect-error
export * from 'vue'
