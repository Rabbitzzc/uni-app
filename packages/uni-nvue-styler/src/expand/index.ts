import { extend } from '@vue/shared'
import type { Plugin } from 'postcss'
import { NormalizeOptions, TransformDecl } from '../utils'
import { transformBackground } from './background'
import { transformBorder } from './border'
import { transformBorderColor } from './borderColor'
import { transformBorderRadius } from './borderRadius'
import { transformBorderStyle } from './borderStyle'
import { transformBorderWidth } from './borderWidth'
import { transformFlexFlow } from './flexFlow'
import { transformFont } from './font'
import { transformMargin } from './margin'
import { transformPadding } from './padding'
import { transformTransition } from './transition'

function getDeclTransforms(
  options: NormalizeOptions
): Record<string, TransformDecl> {
  const result: Record<string, TransformDecl> = {
    transition: transformTransition,
    border: transformBorder,
    background: transformBackground,
  }
  // TODO transtion 简写属性
  if (options.type !== 'uvue') {
    extend(result, {
      margin: transformMargin,
      padding: transformPadding,
    })
  }
  if (__NODE_JS__) {
    extend(result, {
      'border-top': transformBorder,
      'border-right': transformBorder,
      'border-bottom': transformBorder,
      'border-left': transformBorder,
      'border-style': transformBorderStyle,
      'border-width': transformBorderWidth,
      'border-color': transformBorderColor,
      'border-radius': transformBorderRadius,
      'flex-flow': transformFlexFlow,
      font: transformFont,
    })
  } else {
    extend(result, {
      borderTop: transformBorder,
      borderRight: transformBorder,
      borderBottom: transformBorder,
      borderLeft: transformBorder,
      borderStyle: transformBorderStyle,
      borderWidth: transformBorderWidth,
      borderColor: transformBorderColor,
      borderRadius: transformBorderRadius,
      flexFlow: transformFlexFlow,
    })
  }
  return result
}

let DeclTransforms: Record<string, TransformDecl>
const expanded = Symbol('expanded')
export function expand(options: NormalizeOptions): Plugin {
  const plugin: Plugin = {
    postcssPlugin: 'nvue:expand',
    Declaration(decl) {
      if ((decl as any)[expanded]) {
        return
      }
      if (!DeclTransforms) {
        DeclTransforms = getDeclTransforms(options)
      }
      const transform = DeclTransforms[decl.prop]
      if (transform) {
        const res = transform(decl)
        const isSame = res.length === 1 && res[0] === decl
        if (!isSame) {
          decl.replaceWith(res)
        }
      }
      ;(decl as any)[expanded] = true
    },
  }
  return plugin
}
