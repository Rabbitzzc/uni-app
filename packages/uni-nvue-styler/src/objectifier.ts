import type { Container, Document, Root } from 'postcss'
import { extend, hasOwn } from '@vue/shared'
// import { COMBINATORS_RE } from './utils'
import parser from 'postcss-selector-parser'

interface ObjectifierContext {
  'FONT-FACE': Record<string, unknown>[]
  TRANSITION: Record<string, Record<string, unknown>>
}

export function objectifier(
  node: Root | Document | Container | null,
  { trim }: { trim: boolean } = { trim: false }
) {
  if (!node) {
    return {}
  }
  const context: ObjectifierContext = {
    'FONT-FACE': [],
    TRANSITION: {},
  }
  const result = transform(node, context)
  if (trim) {
    trimObj(result)
  }
  if (context['FONT-FACE'].length) {
    result['@FONT-FACE'] = context['FONT-FACE']
  }
  if (Object.keys(context.TRANSITION).length) {
    result['@TRANSITION'] = context.TRANSITION
  }
  return result
}

function trimObj(obj: Record<string, any>) {
  Object.keys(obj).forEach((name) => {
    const value = obj[name]
    if (Object.keys(value).length === 1 && hasOwn(value, '')) {
      obj[name] = value['']
    }
  })
}

function transform(
  node: Root | Document | Container,
  context: ObjectifierContext
) {
  const result: Record<string, Record<string, unknown> | unknown> = {}
  node.each((child) => {
    if (child.type === 'atrule') {
      const body = transform(child, context)
      const fontFamily = body.fontFamily as string
      if (fontFamily && '"\''.indexOf(fontFamily[0]) > -1) {
        body.fontFamily = fontFamily.slice(1, fontFamily.length - 1)
      }
      context['FONT-FACE'].push(body)
    } else if (child.type === 'rule') {
      const body = transform(child, context)
      child.selectors.forEach((selector) => {
        transformSelector(selector, body, result, context)
      })
    } else if (child.type === 'decl') {
      if (child.important) {
        result['!' + child.prop] = child.value
        // !important的值域优先级高，故删除非!important的值域
        delete result[child.prop]
      } else {
        if (!hasOwn(result, '!' + child.prop)) {
          result[child.prop] = child.value
        }
      }
    }
  })
  return result
}
const processer = parser()
function transformSelector(
  selector: string,
  body: Record<string, unknown>,
  result: Record<string, unknown | Record<string, unknown>>,
  context: ObjectifierContext
) {
  const ans = processer.astSync(selector)
  let sele = ans.nodes[0]
  const currSel = parser.selector({ value: '' })
  let lastCombinator = -1
  // 定位最后一个组合选择器
  for (let i = sele.nodes.length - 1; i >= 0; i--) {
    if (sele.nodes[i].type === 'combinator') {
      lastCombinator = i
      break
    }
  }
  // 添加组合选择器的下一个选择器
  currSel.append(sele.nodes.splice(lastCombinator + 1, 1) as any)
  if (
    sele.nodes[lastCombinator + 1] &&
    sele.nodes[lastCombinator + 1].type === 'pseudo'
  ) {
    // 添加伪类选择器
    currSel.append(sele.nodes.splice(lastCombinator + 1, 1) as any)
  }
  let parentSelector = ans.toString()
  // .a:checked => a:checked
  let curSelector = currSel.toString().substring(1)

  const pseudoIndex = curSelector.indexOf(':')
  if (pseudoIndex > -1) {
    const pseudoClass = curSelector.slice(pseudoIndex)
    curSelector = curSelector.slice(0, pseudoIndex)
    Object.keys(body).forEach(function (name) {
      body[name + pseudoClass] = body[name]
      delete body[name]
    })
  }
  transition(curSelector, body, context)
  if (!Object.keys(body).length) {
    return
  }
  result = (result[curSelector] || (result[curSelector] = {})) as Record<
    string,
    unknown
  >

  if (result[parentSelector]) {
    // clone
    result[parentSelector] = processImportant(
      extend({}, result[parentSelector], body)
    )
  } else {
    result[parentSelector] = body
  }
}

/**
 * 处理 important 属性，如果某个属性是 important，需要将非 important 的该属性移除掉
 * @param body
 */
function processImportant(body: Record<string, unknown>) {
  Object.keys(body).forEach((name) => {
    if (name.startsWith('!')) {
      delete body[name.substring(1)]
    }
  })
  return body
}

function transition(
  className: string,
  body: Record<string, unknown>,
  { TRANSITION }: ObjectifierContext
) {
  Object.keys(body).forEach((prop) => {
    if (prop.indexOf('transition') === 0 && prop !== 'transition') {
      const realProp = prop.replace('transition', '')
      TRANSITION[className] = TRANSITION[className] || {}
      TRANSITION[className][realProp[0].toLowerCase() + realProp.slice(1)] =
        body[prop]
    }
  })
}
