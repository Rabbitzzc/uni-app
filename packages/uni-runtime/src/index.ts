export { isFunction, isString } from '@vue/shared'
export const __uniConfig = globalThis.__uniConfig
