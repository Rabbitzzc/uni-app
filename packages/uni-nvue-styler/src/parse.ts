import postcss, { Message } from 'postcss'
import { objectifier } from './objectifier'
import { expand } from './expand'
import { normalize } from './normalize'
import { NormalizeOptions } from './utils'

interface ParseOptions extends NormalizeOptions {
  filename?: string
  map?: boolean
  mapOf?: boolean
  ts?: boolean
  chunk?: number
  noCode?: boolean
}

export async function parse(input: string, options: ParseOptions = {}) {
  const { root, messages } = await postcss([
    expand(options),
    normalize(options),
  ])
    .process(input, {
      from: options.filename || 'foo.css',
    })
    .catch((err: any) => {
      return {
        root: null,
        messages: [
          {
            type: 'error',
            text: err.message,
          } as Message,
        ],
      }
    })
  if (options.noCode === true) {
    return { code: '', messages }
  }
  const obj = root ? objectifier(root) : {}
  if (options.map || options.mapOf) {
    return {
      code: mapToInitStringChunk(
        objToMap(obj),
        options.ts,
        true,
        options.mapOf,
        options.chunk
      ),
      messages,
    }
  }
  return { code: JSON.stringify(obj), messages }
}

function mapToInitStringChunk(
  map: Map<string, unknown>,
  ts: boolean = false,
  isRoot: boolean = false,
  isMapOf: boolean = false,
  chunk: number = 0
): string {
  if (!chunk) {
    return mapToInitString(map, ts, isRoot, isMapOf)
  }
  const chunks: string[] = []
  let chunkMap: Map<string, unknown> = new Map()
  let chunkCount = 0
  for (const [key, value] of map) {
    if (chunkCount === chunk) {
      chunks.push(mapToInitString(chunkMap, ts, isRoot, isMapOf))
      chunkMap = new Map()
      chunkCount = 0
    }
    chunkMap.set(key, value)
    chunkCount++
  }
  if (chunkCount) {
    chunks.push(mapToInitString(chunkMap, ts, isRoot, isMapOf))
  }
  return `[${chunks.join(',')}]`
}

function mapToInitString(
  map: Map<string, unknown>,
  ts: boolean = false,
  isRoot: boolean = false,
  isMapOf: boolean = false
): string {
  const entries = []
  for (let [key, value] of map) {
    if (value instanceof Map) {
      entries.push(`["${key}", ${mapToInitString(value, ts, false, isMapOf)}]`)
    } else {
      entries.push(`["${key}", ${JSON.stringify(value)}]`)
    }
  }
  if (isMapOf) {
    return `utsMapOf([${entries.join(', ')}])`
  }
  return `new Map${
    ts
      ? isRoot
        ? '<string, Map<string, Map<string, any>>>'
        : '<string, any>'
      : ''
  }([${entries.join(', ')}])`
}

function objToMap(obj: Record<string, unknown>) {
  const map = new Map()
  for (const key in obj) {
    const value = obj[key]
    if (typeof value === 'object') {
      map.set(key, objToMap(value as Record<string, unknown>))
    } else {
      map.set(key, value)
    }
  }
  return map
}
