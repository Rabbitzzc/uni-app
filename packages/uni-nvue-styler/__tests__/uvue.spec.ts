import { parse } from '../src/index'
describe('uvue-style', () => {
  test('js', async () => {
    const { code } = await parse(
      `
        .content {
            display: flex;
        } 
        .content .logo {
            width: 200rpx;
            height: 200rpx;
        }
        .text-area, .title {
            font-size: 36rpx;
        }
        `,
      { map: true }
    )
    expect(code).toMatchSnapshot()
  })
  test('ts', async () => {
    const { code } = await parse(
      `
        .content {
            display: flex;
        } 
        .content .logo {
            width: 200rpx;
            height: 200rpx;
        }
        .text-area, .title {
            font-size: 36rpx;
        }
        `,
      { map: true, ts: true }
    )
    expect(code).toMatchSnapshot()
  })
  test('chunk', async () => {
    const { code } = await parse(
      `
        .content {
            display: flex;
        } 
        .content .logo {
            width: 200rpx;
            height: 200rpx;
        }
        .text-area, .title {
            font-size: 36rpx;
        }
        `,
      { map: true, ts: true, chunk: 2 }
    )
    expect(code).toMatchSnapshot()
  })
  test('chunk with mapOf', async () => {
    const { code } = await parse(
      `
        .content {
            display: flex;
        } 
        .content .logo {
            width: 200rpx;
            height: 200rpx;
        }
        .text-area, .title {
            font-size: 36rpx;
        }
        `,
      { mapOf: 'utsMapOf', chunk: 2 }
    )
    expect(code).toMatchSnapshot()
  })
  test('chunk font-face', async () => {
    const { code } = await parse(
      `
        @font-face { 
          font-family: "font-family-name-1"; 
          src: url("font file url 1-1") format("truetype");
        }
        .content {
            display: flex;
        } 
        .content .logo {
            width: 200rpx;
            height: 200rpx;
        }
        .text-area, .title {
            font-size: 36rpx;
            transition-property: margin-top; 
            transition-duration: 300ms;
        }
        `,
      { map: true, ts: true, chunk: 2 }
    )
    expect(code).toMatchSnapshot()
  })
  test('chunk font-face with mapOf', async () => {
    const { code } = await parse(
      `
        @font-face { 
          font-family: "font-family-name-1"; 
          src: url("font file url 1-1") format("truetype");
        }
        .content {
            display: flex;
        } 
        .content .logo {
            width: 200rpx;
            height: 200rpx;
        }
        .text-area, .title {
            font-size: 36rpx;
            transition-property: margin-top; 
            transition-duration: 300ms;
        }
        `,
      { mapOf: 'utsMapOf', chunk: 2 }
    )
    expect(code).toMatchSnapshot()
  })
  test('chunk font-face with mapOf and trim', async () => {
    const { code } = await parse(
      `
        @font-face { 
          font-family: "font-family-name-1"; 
          src: url("font file url 1-1") format("truetype");
        }
        .content {
            display: flex;
        } 
        .content .logo {
            width: 200rpx;
            height: 200rpx;
        }
        .text-area, .title {
            font-size: 36rpx;
            transition-property: margin-top; 
            transition-duration: 300ms;
        }
        `,
      { mapOf: 'utsMapOf', chunk: 2, trim: true }
    )
    expect(code).toMatchSnapshot()
  })
})
