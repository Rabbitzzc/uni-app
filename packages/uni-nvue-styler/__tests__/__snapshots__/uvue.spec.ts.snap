// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`uvue-style chunk 1`] = `"[new Map<string, Map<string, Map<string, any>>>([["content", new Map<string, any>([["", new Map<string, any>([["display", "flex"]])]])], ["logo", new Map<string, any>([[".content ", new Map<string, any>([["width", "200rpx"], ["height", "200rpx"]])]])]]),new Map<string, Map<string, Map<string, any>>>([["text-area", new Map<string, any>([["", new Map<string, any>([["fontSize", "36rpx"]])]])], ["title", new Map<string, any>([["", new Map<string, any>([["fontSize", "36rpx"]])]])]])]"`;

exports[`uvue-style chunk font-face 1`] = `"[new Map<string, Map<string, Map<string, any>>>([["content", new Map<string, any>([["", new Map<string, any>([["display", "flex"]])]])], ["logo", new Map<string, any>([[".content ", new Map<string, any>([["width", "200rpx"], ["height", "200rpx"]])]])]]),new Map<string, Map<string, Map<string, any>>>([["text-area", new Map<string, any>([["", new Map<string, any>([["fontSize", "36rpx"], ["transitionProperty", "marginTop"], ["transitionDuration", 300]])]])], ["title", new Map<string, any>([["", new Map<string, any>([["fontSize", "36rpx"], ["transitionProperty", "marginTop"], ["transitionDuration", 300]])]])]]),new Map<string, Map<string, Map<string, any>>>([["@FONT-FACE", new Map<string, any>([["0", new Map<string, any>([["fontFamily", "font-family-name-1"], ["src", "url(\\"font file url 1-1\\") format(\\"truetype\\")"]])]])], ["@TRANSITION", new Map<string, any>([["text-area", new Map<string, any>([["property", "marginTop"], ["duration", 300]])], ["title", new Map<string, any>([["property", "marginTop"], ["duration", 300]])]])]])]"`;

exports[`uvue-style chunk font-face with mapOf 1`] = `"[utsMapOf([["content", utsMapOf([["", utsMapOf([["display", "flex"]])]])], ["logo", utsMapOf([[".content ", utsMapOf([["width", "200rpx"], ["height", "200rpx"]])]])]]),utsMapOf([["text-area", utsMapOf([["", utsMapOf([["fontSize", "36rpx"], ["transitionProperty", "marginTop"], ["transitionDuration", 300]])]])], ["title", utsMapOf([["", utsMapOf([["fontSize", "36rpx"], ["transitionProperty", "marginTop"], ["transitionDuration", 300]])]])]]),utsMapOf([["@FONT-FACE", utsMapOf([["0", utsMapOf([["fontFamily", "font-family-name-1"], ["src", "url(\\"font file url 1-1\\") format(\\"truetype\\")"]])]])], ["@TRANSITION", utsMapOf([["text-area", utsMapOf([["property", "marginTop"], ["duration", 300]])], ["title", utsMapOf([["property", "marginTop"], ["duration", 300]])]])]])]"`;

exports[`uvue-style chunk font-face with mapOf and trim 1`] = `"[utsMapOf([["content", padStyleMapOf(utsMapOf([["display", "flex"]]))], ["logo", utsMapOf([[".content ", utsMapOf([["width", "200rpx"], ["height", "200rpx"]])]])]]),utsMapOf([["text-area", padStyleMapOf(utsMapOf([["fontSize", "36rpx"], ["transitionProperty", "marginTop"], ["transitionDuration", 300]]))], ["title", padStyleMapOf(utsMapOf([["fontSize", "36rpx"], ["transitionProperty", "marginTop"], ["transitionDuration", 300]]))]]),utsMapOf([["@FONT-FACE", utsMapOf([["0", utsMapOf([["fontFamily", "font-family-name-1"], ["src", "url(\\"font file url 1-1\\") format(\\"truetype\\")"]])]])], ["@TRANSITION", utsMapOf([["text-area", utsMapOf([["property", "marginTop"], ["duration", 300]])], ["title", utsMapOf([["property", "marginTop"], ["duration", 300]])]])]])]"`;

exports[`uvue-style chunk with mapOf 1`] = `"[utsMapOf([["content", utsMapOf([["", utsMapOf([["display", "flex"]])]])], ["logo", utsMapOf([[".content ", utsMapOf([["width", "200rpx"], ["height", "200rpx"]])]])]]),utsMapOf([["text-area", utsMapOf([["", utsMapOf([["fontSize", "36rpx"]])]])], ["title", utsMapOf([["", utsMapOf([["fontSize", "36rpx"]])]])]])]"`;

exports[`uvue-style js 1`] = `"new Map([["content", new Map([["", new Map([["display", "flex"]])]])], ["logo", new Map([[".content ", new Map([["width", "200rpx"], ["height", "200rpx"]])]])], ["text-area", new Map([["", new Map([["fontSize", "36rpx"]])]])], ["title", new Map([["", new Map([["fontSize", "36rpx"]])]])]])"`;

exports[`uvue-style ts 1`] = `"new Map<string, Map<string, Map<string, any>>>([["content", new Map<string, any>([["", new Map<string, any>([["display", "flex"]])]])], ["logo", new Map<string, any>([[".content ", new Map<string, any>([["width", "200rpx"], ["height", "200rpx"]])]])], ["text-area", new Map<string, any>([["", new Map<string, any>([["fontSize", "36rpx"]])]])], ["title", new Map<string, any>([["", new Map<string, any>([["fontSize", "36rpx"]])]])]])"`;
